docs/design_docs/user_facing_api.md

```diff
@@ -0,0 +1,24 @@
+# Proposal for user facing API
+
+## Objectives
+- deploy the plugins without day 1 configuration
+- dynamic day 2 host resource exposure
+
+## Requirements
+- keep the DP as independent from multus and CNI as possible
+- keep the explicit configuration (configMap)
+- have an implicit configuration (would be overriden by the explicit)
+
+## Alternatives
+
+### Expose all host interfaces
+This approach would be valid for MVP.
+
+It would:
+  - read data from configMap
+  - if configMap doesn't exist
```

Comment:
This could provide unexpected results.

On day-1, I would have resources for all the NICs and bonds. Let's say I used them for some VMs. On day-2, I create an explicit configuration and loose all the pre-existing resources. The whitelist-based explicit API does not play well with this.

An alternative would be to always expose all NICs and bonds as resources under their respective names. Explicit API would add "logical" resources, where we for instance select all interfaces connected to specific network under name "red".

Reply:
> It would be interesting to allow fine control of disabling some or all of the default resources. Options are:
> 
> * White-list approach: An explicit configuration could include any default resource already in use so that nothing really changes at the end. In my mind this would be ok but I agree how it could be seen by others as counter intuitive and probably catching off guard the unwary.

This is not domain-specific, it's a general problem that happens whenever "stuff" is removed from a white-list, and that "stuff" is being used. 

I think its not far-fetched to expect the admin to **not** remove resources that the stuff he administers (is this even a word ? 0_o) is using. If we were speaking of different roles... I probably would have a different opinion.

> * Global option to disable all default resources.

I'm probably short sighted, but I don't see how the global option would be used. Would it fall back to the explicit configuration ? Feels anti-natural. 

> * Blacklist option to disable some of the default resources.

I tend to not like blacklists, but given what we're trying to achieve, I don't have any argument against this ... I feel that 'exposing' all host interfaces for every user is too much; giving them the ability to protect selected interfaces should be enough.  

I still do prefer the white-list option though.

Reply:
> I think its not far-fetched to expect the admin to **not** remove resources that the stuff he administers (is this even a word ? 0_o) is using. If we were speaking of different roles... I probably would have a different opinion.

It's because going from an initial situation of no configuration applied, and then setting a configuration could be perceived as **adding** and not **removing**. Usually empty white-lists do not allow anything, while our empty config allows/exposes default resources. This perception could be changed if instead of talking about empty configuration, we talked about a default configuration, something like`[{ "*", "*", "*", "*" }]`

> I'm probably short sighted, but I don't see how the global option would be used. Would it fall back to the explicit configuration ? Feels anti-natural.

It could work like this:
* global option unset + no explicit config -> default resources exposed
* global option set + no explicit config -> no resources exposed/invalid config
* global option unset + explicit config -> default + config resources exposed
* global option set + explicit config -> config resources exposed